exercises/giovanni_disiena/ex02_oo_basics/src/ShapeBase.h

```diff
@@ -0,0 +1,21 @@
+#ifndef SHAPEBASE_H
+#define SHAPEBASE_H
+
+#include <math.h>
+#include <string>
+
+class ShapeBase {
+public:
+  // constructs and initialises ShapeBase object
+  ShapeBase(unsigned int, std::string);
+  // functions to get member variables
+  virtual double getPerimeter() const = 0;
+  virtual double getArea() const = 0;
+  std::string getName() const;
+  unsigned int getSides() const;
+private:
+  unsigned int m_sides;
```

Comment:
It would be clearer if this was called m_numberOfSides or something similar as if someone else was using it they might think this was the length of the sides or something